mlflow/store/dbmodels/initial_artifact_store_models.py

```diff
@@ -0,0 +1,66 @@
+# Snapshot of MLflow DB models as of the 0.9.1 release, prior to the first database migration.
+# Used to standardize initial database state.
+# Copied with modifications from
+# https://github.com/mlflow/mlflow/blob/v0.9.1/mlflow/store/dbmodels/models.py, which
+# is the first database schema that users could be running. In particular, modifications have
+# been made to substitute constants from MLflow with hard-coded values (e.g. replacing
+# SourceType.to_string(SourceType.NOTEBOOK) with the constant "NOTEBOOK") and ensure
+# that all constraint names are unique. Note that pre-1.0 database schemas did not have unique
+# constraint names - we provided a one-time migration script for pre-1.0 users so that their
+# database schema matched the schema in this file.
+
+from sqlalchemy import (
+    Column, String, VARBINARY, BigInteger, Integer, PrimaryKeyConstraint)
+from sqlalchemy.ext.declarative import declarative_base
+
+from mlflow.entities import FileInfo
+import os
+
+Base = declarative_base()
+
+
+class SqlArtifact(Base):
+    """
+    DB model for :py:class:`mlflow.entities.Artifact`. These are recorded in ``artifact`` table.
+    """
+    __tablename__ = 'artifacts'
+    artifact_id = Column(Integer, autoincrement=True)
+    """
+    Artifact ID: `Integer`. *Primary Key* for ``artifact`` table.
+    """
+    artifact_name = Column(String(256), nullable=False)
+    """
+    Artifact Name: ``String` (limit 256 characters).
+    """
+    group_path = Column(String(256), nullable=False)
+    """
+    Group path: `String` (limit 256 characters).
+    """
+    artifact_content = Column(VARBINARY, nullable=False)
+    """
+    Artifact : `VarBinary`. Defined as *Non null* in table schema.
+    """
+    artifact_initial_size = Column(BigInteger, nullable=True)
+    """
+    Artifact Initial Size : `BigInteger`. Defined as *null* in table schema.
+    """
+    __table_args__ = (
+        PrimaryKeyConstraint('artifact_id', name='artifact_pk'),
+    )
+
+    def to_file_info(self):
+        """
+        Convert DB model to corresponding FileInfo object.
+
+        :return: :py:class:`mlflow.entities.FileInfo`.
+        """
+        return FileInfo(
+            path=os.path.join(self.group_path, self.artifact_name),
```

Comment:
```suggestion
            path=self.artifact_name,
```

This will allow the `FileInfo` results to conform to the expected semantics of the `ListArtifacts` API. MLflow expects `list_artifacts` to return file paths *relative* to the specified input path.

Reply:
@dbzcumar, I'm a bit confused with this comment, Assume the full path is
db_uri/runid/artifacts/subir/filename

What part of it should be captured in the fileinfo object? With the change you are suggesting fileinfo will just contain the filename part (not the subdir). Is that the expected behavior?

Reply:
@avflor Correct, `path` should only contain the filename part.

Reply:
@dbczumar Thanks for your prompt reply, but I'm still not sure this is the correct behavior. As you said the `FileInfo `is returned by `list_artifacts` which is in turn invoked by `download_artifacts`. If you take a look at the `download_artifacts` in artifact_repo.py  it expects as input a relative source path to the desired artifact ( I assume that this means relative to the root). This function invokes the `download_artifacts_into` function which also takes as input a relative path. The latter function is recursive and every time it is invoked the relative path is instantiated by the FileInfo object. It seems that by passing just the filename in this object this download will not work. Am I missing sth here?

Reply:
@avflor Sorry for the confusion. You are correct. The current behavior of `to_file_info()` is correct. Unfortunately, the UI is unable to display artifacts. I'll spend some time trying to debug that issue independently of `to_file_info()`.

Reply:
@dbczumar Sounds good. Is the ui problem only related to this PR or do you see issues with other types of artifact repos? I'm thinking whether the problem might be on how we define "relative_path" in the context of db_artifact_repo. For example if the full path is:

 db_uri/runid/artifacts/subdir/filename

my Fileindo currently returns "run_id/artifacts/subdir/filename". I wonder whether it should return "subdir/filename".

Reply:
@avflor The UI problem only seems to occur when using `DBArtifactRepository` from this PR. I think you've correctly identified the issue. `FileInfo` should return `subdir/filename`. One possible solution is to add a `base_path` argument to `to_file_info()` as follows:
```
    def to_file_info(self, base_path):
        """
        Convert DB model to corresponding FileInfo object.

        :return: :py:class:`mlflow.entities.FileInfo`.
        """
        return FileInfo(
            path=os.path.relpath(path=os.path.join(self.group_path, self.artifact_name), start=base_path),
            is_dir=False,
            file_size=self.artifact_initial_size)
```
You can then invoke `to_file_info(path)` in `list_artifacts`:
```
        path = os.path.join(self.root, path) if path is not None else self.root
        with self.ManagedSessionMaker() as session:
            return [artifact.to_file_info(path)
                    for artifact in
                    session.query(SqlArtifact).filter(
                        SqlArtifact.group_path.like(
                            path + "%"))]
```

Thanks for catching this!

Reply:
It seems that with this change the support of None in the `list_artifacts`, the UI works now.